src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -27,15 +34,16 @@ class Debugger {
 
     readonly session: Promise<DebugSession>;
 
-    constructor(public readonly document: NotebookDocument) {
+    constructor(public readonly document: NotebookDocument, public readonly cell?: NotebookCell) {
+        const name = cell ? `${document.uri.toString()}?RBL=${cell.index}` : document.uri.toString();
         this.session = new Promise<DebugSession>((resolve, reject) => {
             this.resolveFunc = resolve;
             this.rejectFunc = reject;
 
             debug
                 .startDebugging(undefined, {
                     type: DataScience.pythonKernelDebugAdapter(),
-                    name: `${path.basename(document.uri.toString())}`,
+                    name: `${path.basename(name)}`,
```

Comment:
path.basename works with ```?``` in it? I mean I'd think that would work but seems like an invalid file name?

Reply:
it does work, I use it to differentiate between run by line and normal debugging.

I'm open to change it if you have a suggestion

Reply:
nvm, I'll use it the same, but will use path.basename only on the doc URI